### PR TITLE
Feature reg

### PR DESCRIPTION
**Pull request by andela-sjames**
ready to merge

### CODE REVIEW

**Comment by andela-sjames** [comment]
ready to merge

File: troupon/account/forms.py [hunk 1]
```diff
@@ -0,0 +1,29 @@
+from django import forms
+from django.contrib.auth.models import User
+from django.contrib.auth.forms import UserCreationForm
+ 
+class MySignupForm(UserCreationForm):
```
Comment (andela-cnnadi) [inline troupon/account/forms.py]: Improper name for form

File: troupon/account/forms.py [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+    '''
+    fields defined to override default field property
+    '''
+    email = forms.EmailField(required = True)
+    first_name = forms.CharField(required = True)
+    last_name = forms.CharField(required=True)
+
+
+    class meta:
```
Comment (andela-cnnadi) [inline troupon/account/forms.py]: Isn't this supposed to be capital 'M'?

File: troupon/account/tests/testreg.py [hunk 3]
```diff
@@ -0,0 +1,44 @@
+from django.template.loader import render_to_string
+from django.test import TestCase, Client
+from django.core.context_processors import csrf
+
+class RegViewTest(TestCase):
```
Comment (andela-cnnadi) [inline troupon/account/tests/testreg.py]: Follow the file naming convention for tests. See Femi

File: troupon/account/urls.py [hunk 4]
```diff
@@ -1,6 +1,12 @@
-from django.conf.urls import include, url
+from django.conf.urls import patterns, include, url
 from account.views import UserSigninView
+from account import views
 
-urlpatterns = [
+urlpatterns = patterns('',
+    url(r'^signup/$',views.UserSignupView.as_view(),name = 'UserSignupView'),
+    url(r'^signupreq/$',views.UserSignupreq.as_view(),name = 'UserSignupreq'),
```
Comment (andela-cnnadi) [inline troupon/account/urls.py]: Variable names. signupreq?

File: troupon/account/views.py [hunk 5]
```diff
@@ -89,3 +93,45 @@ def some_view(request):
         # add the slash at the relative path's view and finished
         referer = u'/' + u'/'.join(referer[1:])
         return referer
+
+class UserSignupRequest(View):
```
Comment (andela-cnnadi) [inline troupon/account/views.py]: A lot of repetition going on here. Can you see if you can implement this using `FormView`.

File: troupon/account/views.py [hunk 6]
```diff
@@ -89,3 +93,45 @@ def some_view(request):
...
+        if usersignupform.is_valid():
+            print ('form is valid')
+            usersignupform.save()
+
+            return HttpResponseRedirect('/account/confirm/')
+
+        else:
+          return HttpResponseRedirect('/account/signup/')
+ 
+class UserSignupView(TemplateView):
```
Comment (andela-cnnadi) [inline troupon/account/views.py]: Not sure why there are two views for SignUp?

File: troupon/account/tests/test_reg.py [hunk 7]
```diff
@@ -0,0 +1,44 @@
+from django.template.loader import render_to_string
```
Comment (andela-cnnadi) [inline troupon/account/tests/test_reg.py]: Embed all this code within test_views.py

File: troupon/troupon/urls.py [hunk 8]
```diff
@@ -3,6 +3,7 @@
 from django.contrib import admin
 import deals
 from account import views
+import account 
 
 urlpatterns = [
     url(r'^auth/', include('account.urls')),
```
Comment (andela-cnnadi) [inline troupon/troupon/urls.py]: Can you change this to account...

**Comment by andela-sjames** [comment]
rebased with develop and ready to merge boss

File: troupon/account/templates/account/signup.html [hunk 9]
```diff
@@ -0,0 +1,58 @@
...
+
+      </div>
+    
+      <div class="col-md-6">
+        <h3 class="dark-grey">Facebook Sign-Up</h3>
+        <p>
+          Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book. It has survived not only five centuries, but also the leap into electronic typesetting, 
+        </p>
+
+<<<<<<< HEAD
```
Comment (andela-cnnadi) [inline troupon/account/templates/account/signup.html]: WTF? Why did you commit a merge conflict? Never ever again please.

File: troupon/account/forms.py [hunk 10]
```diff
@@ -14,8 +17,38 @@ class ResetPasswordForm(forms.Form):
             "class": "",
             "placeholder": ""
         }))
+    class meta:
```
Comment (andela-cnnadi) [inline troupon/account/forms.py]: Use uppercase for the `Meta`

File: troupon/templates/base.html [hunk 11]
```diff
@@ -128,14 +128,15 @@
             </div>            
         </div>
     </nav>
-    <br ><br >
+    <br ><br ><br ><br ><br ><br >
```
Comment (andela-cnnadi) [inline troupon/templates/base.html]: Why multiple breaks?

File: .idea/misc.xml [hunk 12]
```diff
@@ -0,0 +1,14 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (andela-cnnadi) [inline .idea/misc.xml]: lol...You started using PyCharm. This files should be .gitignored.

**Pull request merged**